lib/byron/bench/Restore.hs

```diff
@@ -0,0 +1,468 @@
+    , Block (..)
+    , BlockHeader (..)
+    , BlockchainParameters (..)
+    , ChimericAccount
+    , EpochNo (..)
+    , SlotId (..)
+    , SlotNo (..)
+    , StartTime (..)
+    , SyncProgress (..)
+    , UTxO (..)
+    , WalletId (..)
+    , WalletName (..)
+    , mkSyncTolerance
+    , slotParams
+    , syncProgressRelativeToTime
+    )
+import Cardano.Wallet.Unsafe
+    ( unsafeMkMnemonic, unsafeRunExceptT )
+import Control.Concurrent
+    ( forkIO, threadDelay )
+import Control.DeepSeq
+    ( NFData, rnf )
+import Control.Exception
+    ( bracket, evaluate, throwIO )
+import Control.Monad
+    ( forM, mapM_, void )
+import Control.Monad.Fail
+    ( MonadFail )
+import Control.Monad.IO.Class
+    ( MonadIO (..) )
+import Control.Monad.Trans.Except
+    ( runExceptT )
+import Control.Tracer
+    ( Tracer (..), traceWith )
+import Criterion.Measurement
+    ( getTime, initializeTime, secs )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Quantity
+    ( Quantity (..) )
+import Data.Text
+    ( Text )
+import Data.Time.Clock.POSIX
+    ( POSIXTime, getCurrentTime, getPOSIXTime, utcTimeToPOSIXSeconds )
+import Database.Persist.Sql
+    ( runMigrationSilent )
+import Fmt
+    ( fmt, pretty, (+|), (+||), (|+), (||+) )
+import Ouroboros.Network.NodeToClient
+    ( NodeToClientVersionData (..) )
+import Say
+    ( sayErr )
+import System.Environment
+    ( getEnv )
+import System.FilePath
+    ( (</>) )
+import System.IO
+    ( BufferMode (..)
+    , IOMode (..)
+    , hFlush
+    , hSetBuffering
+    , stderr
+    , stdout
+    , withFile
+    )
+import System.IO.Temp
+    ( withSystemTempFile )
+
+import qualified Cardano.BM.Configuration.Model as CM
+import qualified Cardano.BM.Data.BackendKind as CM
+import qualified Cardano.Wallet as W
+import qualified Cardano.Wallet.DB.Sqlite as Sqlite
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+
+
+-- | Run all available benchmarks. Can accept one argument that is a target
+-- network against which benchmarks below should be ran
+--
+main :: IO ()
+main = do
+    hSetBuffering stdout NoBuffering
+    hSetBuffering stderr NoBuffering
+
+    (_logCfg, tr) <- initBenchmarkLogging Info
+    installSignalHandlers (return ())
+
+    -- TODO: We might want to support testnet via a
+    -- `--benchmark-arguments "testnet"`
+    let network = Mainnet
+
+    -- Reading setup from env vars.
+    --
+    -- If you usually run ./scripts/mainnet.sh in the cardano-node repository,
+    -- it can be useful to set
+    --
+    --     export BYRON_CONFIGS=$CARDANO_NODE_DIR/configuration
+    --     export NODE_DB=$CARDANO_NODE_DIR/db
+    --
+    -- before running this benchmark manually with stack.
+    configs <- getEnv "BYRON_CONFIGS"
+    nodeDB <- getEnv "NODE_DB"
+
+    let genesisHash = "5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb"
```

Comment:
Note that this is already hard-coded in `Cardano.Wallet.Byron.Compatibility`, would be good limit the number of places where this gets written literally. 

```hs
let genesisHash = getGenesisBlockHash $ blockchainParameters @'Mainnet
```

Reply:
885ffee71e4e53fc0113ed19dcd2be872f242c59